samza-core/src/main/java/org/apache/samza/runtime/ContainerLaunchUtil.java

```diff
@@ -151,10 +151,9 @@ private static void run(
 
       if (new JobConfig(config).getJobCoordinatorHighAvailabilityEnabled()) {
         ExecutionContainerIdManager executionContainerIdManager = new ExecutionContainerIdManager(
-            new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetExecutionContainerIdMapping.TYPE));
-        if (executionContainerIdManager != null && execEnvContainerId.isPresent()) {
-          executionContainerIdManager.writeExecutionEnvironmentContainerIdMapping(containerId, execEnvContainerId.get());
-        }
+            new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetExecutionEnvContainerIdMapping.TYPE));
```

Comment:
Can go inside the `ifPresent` block so that we don't create it unless it is available? Why this is optional by the way? Seems like this is required and we should bail out. unless its used in local flow where it isn't populated.